docker/Dockerfile_txsim

```diff
@@ -0,0 +1,48 @@
+# stage 1 Generate celestia-appd Binary
+FROM docker.io/golang:1.20.6-alpine3.17 as builder
+# hadolint ignore=DL3018
+RUN apk update && apk add --no-cache \
+    gcc \
+    git \
+    make \
+    musl-dev
+COPY . /celestia-app
+WORKDIR /celestia-app
+# we need the appd build as we might want to create an account
+# internally for txsimulation
+RUN make build && make txsim-build
+
+# stage 2
+FROM docker.io/alpine:3.18.2
+
+# Read here why UID 10001: https://github.com/hexops/dockerfile/blob/main/README.md#do-not-use-a-uid-below-10000
+ARG UID=10001
+ARG USER_NAME=celestia
+
+ENV CELESTIA_HOME=/home/${USER_NAME}
+
+# hadolint ignore=DL3018
```

Comment:
[no change needed] because this appears copied from another Dockerfile in this repo: https://github.com/rootulp/celestia-app/blob/20c233258d66453028bc77763d424d43277156ba/Dockerfile#L22

but if we're explicitly ignoring a linter rule it's helpful to explain why via a code comment.